### PR TITLE
Added 'worker_type' and 'glue_version' attributes to the module

### PR DESCRIPTION
**Pull request by vijayanandsharma**
##### SUMMARY

Added support to `glue_version` and `worker_type` attributes in the Glue Module

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

aws_glue_job.py

### CODE REVIEW

**Review by RD89 — APPROVED** [review]

**Review by wimnat — APPROVED** [review]
LGTM

**Comment by gravesm** [comment]
@vijayanandsharma thank you for the PR. Before we can merge this could you add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#community-changelogs)?

File: plugins/modules/aws_glue_job.py [hunk 1]
```diff
@@ -75,6 +77,19 @@
     description:
       - The job timeout in minutes.
     type: int
+  glue_version:
+    description:
+      - Glue version determines the versions of Apache Spark and Python that AWS Glue supports
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:82]:
```suggestion
      - Glue version determines the versions of Apache Spark and Python that AWS Glue supports.
```

File: plugins/modules/aws_glue_job.py [hunk 2]
```diff
@@ -75,6 +77,19 @@
...
+    type: str
+  worker_type:
+    description:
+      - The type of predefined worker that is allocated when a job runs. Accepts a value of Standard, G.1X, or G.2X.
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:86]:
```suggestion
      - The type of predefined worker that is allocated when a job runs.
```

File: plugins/modules/aws_glue_job.py [hunk 3]
```diff
@@ -75,6 +77,19 @@
     description:
       - The job timeout in minutes.
     type: int
+  glue_version:
+    description:
+      - Glue version determines the versions of Apache Spark and Python that AWS Glue supports
+    type: str
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:83]:
```suggestion
    type: str
    version_added: 1.5.0
```

File: plugins/modules/aws_glue_job.py [hunk 4]
```diff
@@ -75,6 +77,19 @@
...
+  worker_type:
+    description:
+      - The type of predefined worker that is allocated when a job runs. Accepts a value of Standard, G.1X, or G.2X.
+    choices: [ 'Standard', 'G.1X', 'G.2X' ]
+    type: str
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:88]:
```suggestion
    type: str
    version_added: 1.5.0
```

File: plugins/modules/aws_glue_job.py [hunk 5]
```diff
@@ -75,6 +77,19 @@
...
+  number_of_workers:
+    description:
+      - The number of workers of a defined workerType that are allocated when a job runs.
+    type: int
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:92]:
```suggestion
    type: int
    version_added: 1.5.0
```

**Review by tremble — APPROVED** [review]
Thanks for adding the change-log entry.

A couple of minor issues that I'll merge.

Something which would be really appreciated would be if you could add some integration tests for aws_glue_job.  They live in tests/integration/targets/<module_name> and are just Ansible roles.  It's much better when someone who's familiar with the service writes them.  Given there aren't any integration tests for aws_glue_job I don't think it's fair to require them to get this PR merged.

File: plugins/modules/aws_glue_job.py [hunk 6]
```diff
@@ -345,7 +375,10 @@ def main():
             name=dict(required=True, type='str'),
             role=dict(type='str'),
             state=dict(required=True, choices=['present', 'absent'], type='str'),
-            timeout=dict(type='int')
+            timeout=dict(type='int'),
+            glue_version=dict(type='str'),
+            worker_type=dict(choices=['Standard', 'G.1X', 'G.2X'], type='str'),
+            number_of_workers=dict(type='int')
```
Comment (tremble) [inline plugins/modules/aws_glue_job.py:381]:
```suggestion
            number_of_workers=dict(type='int'),
```

File: changelogs/fragments/370_add_attributes_glue_module.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- aws_glue_job - added ``worker_type`` and ``glue_version`` attributes to the module (https://github.com/ansible-collections/community.aws/pull/370).
```
Comment (tremble) [inline changelogs/fragments/370_add_attributes_glue_module.yaml:3]:
```suggestion
- aws_glue_job - added ``number_of_workers``, ``worker_type`` and ``glue_version`` attributes to the module (https://github.com/ansible-collections/community.aws/pull/370).
```

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
Many thanks for taking the time to open this PR. I'm sorry it's taken so long to get it reviewed and merged.

**Pull request merged**